lib/logstash/inputs/file.rb

```diff
@@ -170,6 +171,7 @@ def register
     require "digest/md5"
     @logger.info("Registering file input", :path => @path)
     @host = Socket.gethostname.force_encoding(Encoding::UTF_8)
+    @settings = defined?(LogStash::SETTINGS) ? LogStash::SETTINGS : nil
```

Comment:
Can you add a comment above this line that says this is specifically for checking if we're on Logstash 5.x?

Also, @settings probably doesn't need to be an instance variable, does it?